src/SignalR/clients/ts/signalr/src/HttpConnection.ts

```diff
@@ -298,7 +298,7 @@ export class HttpConnection implements IConnection {
     }
 
     private async getNegotiationResponse(url: string): Promise<INegotiateResponse> {
-        const headers = {};
+        const headers = this.options.headers || {};
```

Comment:
If the HttpClient is already applying the default headers, why do we have to use the default headers here? If it's just for custom clients, then why change DefaultHttpClient at all?

Reply:
I think it's fine to keep manually adding default headers to every request in order to support custom clients. I'd like to make sure we shallow copy here and elsewhere before adding additional headers.

This just makes the changes to DefaultHttpClient redundant, so we should undo that part of the change.

I know there's been a lot of back and forth. I appreciate you bearing with us @vansergen.

Reply:
@halter73 Fixed: f013d64